resources/lib/playerinfo.py

```diff
@@ -4,83 +4,149 @@
+        # Get asset_id, title and url from api
+        self.path = getInfoLabel('Player.Filenameandpath')
+        ep_id = play_url_to_id(self.path)
+
+        if ep_id.get('video_id'):
+            episode = self.apihelper.get_episodes(video_id=ep_id.get('video_id'), variety='single')[0]
+        elif ep_id.get('whatson_id'):
+            episode = self.apihelper.get_episodes(whatson_id=ep_id.get('whatson_id'), variety='single')[0]
+        elif ep_id.get('video_url'):
+            episode = self.apihelper.get_episodes(video_url=ep_id.get('video_url'), variety='single')[0]
+
+        self.asset_id = self.resumepoints.assetpath_to_id(episode.get('assetPath'))
+        self.title = episode.get('program')
+        self.url = url_to_episode(episode.get('url', ''))
+        self.whatson_id = episode.get('whatsonId')
+
     def onAVStarted(self):  # pylint: disable=invalid-name
         ''' Called when Kodi has a video or audiostream '''
-        log(3, '[PlayerInfo] %d onAVStarted' % self._id)
-        self._stop.clear()
-        self._last_pos = 0
-        self._total = self.getTotalTime()
-        self._path = getInfoLabel('Player.Filenameandpath')
-        tag = self.getVideoInfoTag()
-        self._info(dict(
-            program=to_unicode(tag.getTVShowTitle()),
-            playcount=tag.getPlayCount(),
-            rating=tag.getRating(),
-            path=self._path,
-            runtime=self._total,
-        ))
+        log(3, '[PlayerInfo] %d onAVStarted' % self.thread_id)
+        self.stop.clear()
+        self.last_pos = 0
+        self.total = self.getTotalTime()
+        self.push_upnext()
         Thread(target=self.stream_position, name='StreamPosition').start()
 
-    def onPlayBackStopped(self):  # pylint: disable=invalid-name
-        ''' Called when user stops Kodi playing a file '''
-        log(3, '[PlayerInfo] %d onPlayBackStopped' % self._id)
-        self._info(dict(path=self._path, position=self._last_pos, total=self._total, event='playbackstopped'))
-        self._stop.set()
-
     def onAVChange(self):  # pylint: disable=invalid-name
         ''' Called when Kodi has a video, audio or subtitle stream. Also happens when the stream changes. '''
 
+    def onPlayBackPaused(self):  # pylint: disable=invalid-name
+        ''' Called when user pauses a playing file '''
+        log(3, '[PlayerInfo] %d onPlayBackPaused' % self.thread_id)
+        self.paused = True
+        self.push_position(position=self.last_pos, total=self.total)
+
+    def onPlayBackResumed(self):  # pylint: disable=invalid-name
+        ''' Called when user resumes a paused file or a next playlist item is started '''
+        suffix = 'after pausing' if self.paused else 'after playlist change'
+        log(3, '[PlayerInfo] %d onPlayBackResumed %s' % (self.thread_id, suffix))
+        if not self.paused:
+            self.push_position(position=self.last_pos, total=self.total)
+        self.paused = False
+
```

Comment:
I noticed you removed the onPlaybackSeek(), which is important to ensure we have updated `self.last_pos`.
Especially for doing things like, skipping to the end, or skipping to the start and stopping.
(Which is currently the only way to mark-as-watched, or reset position.

Reply:
I'll add this just to be sure, but I think `onPlayBackEnded` also updates the last position when seeking to the end of the video.

Reply:
Indeed, when the playback ended, but not when it was stopped.
To be clear, I don't want to update the position on seek, just set `self.last_pos` for in case it is used.